cmd/nerdctl/container_create.go

```diff
@@ -205,10 +206,18 @@ func processContainerCreateOptions(cmd *cobra.Command) (opt types.ContainerCreat
 	if err != nil {
 		return
 	}
-	opt.Device, err = cmd.Flags().GetStringSlice("device")
+
+	allDevices, err := cmd.Flags().GetStringSlice("device")
 	if err != nil {
 		return
 	}
+	for _, device := range allDevices {
+		if cdi.IsQualifiedName(device) {
+			opt.CDIDevices = append(opt.CDIDevices, device)
```

Comment:
I guess we want to keep this experimental until the syntax gets accepted in Docker CLI

https://github.com/containerd/nerdctl/blob/main/docs/experimental.md?plain=1
https://github.com/search?q=repo%3Acontainerd%2Fnerdctl+checkExperimental&type=code

Reply:
Can we have an integration test too?

Reply:
Sure. I will add an integration test and also gate this behind an experimental flag.

Reply:
The Docker CLI PR has been merged, so I will pick up this PR again in the next couple of days.

Reply:
@elezar ping?

Reply:
Sorry for the delay. I will look at this again this week.

Reply:
@AkihiroSuda do we still need to make this feature experimental?

Reply:
If it is consistent with Docker's behavior, it does not need to stay experimental

Reply:
It is consistent, but requries an op-in feature flag in the docker config.

The `--cdi-spec-dirs` folder is also not a docker command line option, but I assume that that is ok.